app/api/edlinker/auditoriums.rb

```diff
@@ -18,4 +19,40 @@ class Edlinker::Auditoriums < Grape::API
       end
     end
   end
+
+  namespace :auditoriums do
+    desc 'create task for auditorium'
+    params { use :task_params }
+    post ':auditorium_id/tasks' do
+      current_auditorium = Auditorium.find_by(id: params[:auditorium_id])
+      error!('Auditorium not found') unless current_auditorium
+      current_auditorium.tasks.create(params)
+    end
+
+    get ':auditorium_id/tasks' do
+      current_auditorium = Auditorium.find_by(id: params[:auditorium_id])
+      error!('Auditorium not found') unless current_auditorium
+      current_user = User.find_by(id: params[:user_id])
```

Comment:
add user presence validation and validate his role
Only teacher should be allowed to create tasks